rmw_fastrtps_cpp/src/rmw_client.cpp

```diff
@@ -261,7 +264,8 @@ rmw_destroy_client(rmw_node_t * node, rmw_client_t * client)
     }
     delete info;
   }
-  delete (client);
+  rmw_free(const_cast<char *>(client->service_name));
```

Comment:
If you go through rmw_create_client() successfully, `service_name` is guaranteed to be non-NULL.  So the only way that service_name could be NULL in destroy is if someone hand-allocated a structure, set `service_name` to NULL, and then passed it to destroy (unlikely, but not impossible).  I'm not sure what our policy is in that situation, because quite a few other things can go wrong in that situation.